### ISSUE TITLE
Disability to animate private property of node with @tool script.

### ISSUE THREAD

**Issue by vyshliy** [godotengine/godot#85139]
### Godot version

4.2 rc1

### System information

ubuntu 22.04

### Issue description

Minor issue with animating property of node with @tool script. Private (not exported) properties are available to choose in property track for nodes with @tool script and you can add it to the timeline. However there is no way to actually animate this track. When key is selected there is no value exposed to AnimationTrackKeyEdit. 

![key](https://github.com/godotengine/godot/assets/72736632/dd473a2e-f31b-46fc-9205-54f9de280ec9)

I assume that private properties either should not be displayed among the properties available for animation, or should be animated like other properties available in the editor.

### Steps to reproduce

- Open Node2D scene in MRP and select AnimationPlayer.
- Select Key of private property track. There is no value on it.
- Compare it to exported track's key, which is the same type. It has value and can be animated. 

### Minimal reproduction project

[Animate prop of tool script.zip](https://github.com/godotengine/godot/files/13416739/Animate.prop.of.tool.script.zip)

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot/issues/84156

**Comment by vyshliy** [comment]
> See also:
> 
>     * [Adding a Property Track for to the Animation Player should only show exported properties #84156](https://github.com/godotengine/godot/issues/84156)

Exactly, I didn't noticed this in search. This issue looks like a duplicate. I think it can be closed if there is no new information here.

**Comment by akien-mga** [comment]
Closed the other one as this one clearly identified that `@tool` is part of the problem, and has an MRP.

@KoBeWi debugged the issue in https://github.com/godotengine/godot/issues/84156#issuecomment-1792323722

> I suspect this line:
> https://github.com/godotengine/godot/blob/5ee983188de97ae027f9b9c1443438063f708a7e/editor/property_selector.cpp#L188
> I didn't test, but I think `@tool` will add `PROPERTY_USAGE_SCRIPT_VARIABLE` to the properties. I wonder if it's needed here.
